src/services/dealer.ts

```diff
@@ -13,8 +13,16 @@ import { DealerPromotion } from "../types/models/dealerPromotion"
 import { WithValidationError } from "../types/withValidationError"
 import { withTokenRefresh } from "../tokenRefresh"
 
-export const fetchDealer = async (id: number): Promise<Dealer> => {
-  return fetchPath(Service.DEALER, `dealers/${id}`)
+import toQueryString from "../lib/toQueryString"
+
+export const fetchDealer = async (
+  id: number,
+  options: { language?: "de" | "en" | "fr" | "it" } = {}
+): Promise<Dealer> => {
+  return fetchPath(
+    Service.DEALER,
+    `dealers/${id}${options.language ? `?${toQueryString(options)}` : ""}`
```

Comment:
sooo if we pass a language, we convert all options to a query string? this will result in other fields than just the language being included

1. if you want to be explicit about passing the options, only check and pass the language option
2. if you want to be flexible, convert all options to query string and append it always when there is any option

Reply:
Negative. The options are typed as `{ language?: "en" | "de" | "fr" | "it" }`.
We simply can't pass more options, typechecker will forbid